binder/src/main/java/io/grpc/binder/internal/BinderServer.java

```diff
@@ -119,14 +121,11 @@ public SocketAddress getListenSocketAddress() {
   public synchronized void shutdown() {
     if (!shutdown) {
       shutdown = true;
-      if (useSharedTimer) {
-        // TODO: Transports may still be using this resource. They should 
-        // be managing its use as well.
-        SharedResourceHolder.release(GrpcUtil.TIMER_SERVICE, executorService);
-      }
       // Break the connection to the binder. We'll receive no more transactions.
       hostServiceBinder.detach();
       listener.serverShutdown();
+      executorServicePool.returnObject(executorService);
+      executorService = null;
```

Comment:
FYI, returnObject always returns `null` for convenience to allow `executorService = executorServicePool.returnObject(executorService);`. But there's nothing wrong with this.

Reply:
Done.